### ISSUE TITLE
Add rule: no-async-foreach

### ISSUE THREAD

**Issue by pertrai1** [pertrai1/eslint-plugin-llm-core#8]
## Problem

LLMs use `forEach` with async callbacks, not realizing forEach doesn't await.

## Rule

Disallow async functions passed to forEach.

### ❌ Bad
```ts
items.forEach(async (item) => {
  await processItem(item); // Not awaited!
});
```

### ✅ Good
```ts
for (const item of items) {
  await processItem(item);
}
```

## Why it matters

- forEach with async silently swallows promises
- Common LLM mistake
- Leads to race conditions
